### PR TITLE
improves json encoder

### PR DESCRIPTION
**Pull request by onacitarhan17**

### CODE REVIEW

File: src/main/java/model/crypto/Sha3256Hash.java [hunk 1]
```diff
@@ -8,7 +9,7 @@
  * Represents SHA3-256 data type which extends abstract Hash data type for
  * the cryptographic hash function used in LightChain.
  */
-public class Sha3256Hash extends Hash {
+public class Sha3256Hash extends Hash implements Serializable {
```
Comment (thep2p) [inline src/main/java/model/crypto/Sha3256Hash.java]: `Sha3256Hash` extends `Hash` which in turn implements `Serializable`. Do we need to re-implement `Serializable` here again when we already have it at the superclass?

File: src/main/java/model/crypto/Signature.java [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import java.io.Serializable;
+
 import model.Entity;
 import model.lightchain.Identifier;
 
 /**
  * Represents abstract data type for the cryptographic digital signature used in LightChain.
  */
-public abstract class Signature extends Entity {
+public abstract class Signature extends Entity implements Serializable {
```
Comment (thep2p) [inline src/main/java/model/crypto/Signature.java]: `Signature` extends `Entity` which in turn implements `Serializable`. Do we need to re-implement `Serializable` here again when we already have it at the superclass?

File: src/main/java/modules/ads/AuthenticatedEntity.java [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+import java.io.Serializable;
+
 import model.Entity;
 
 /**
  * AuthenticatedEntity is a wrapper model around the Entity type that also contains a membership Merkle Proof for
  * that entity against a root identifier.
  */
-public abstract class AuthenticatedEntity extends Entity {
+public abstract class AuthenticatedEntity extends Entity implements Serializable {
```
Comment (thep2p) [inline src/main/java/modules/ads/AuthenticatedEntity.java]: `AuthenticatedEntity` extends `Entity` which in turn implements `Serializable`. Do we need to re-implement `Serializable` here again when we already have it at the superclass?

File: src/main/java/model/crypto/Sha3256Hash.java (hunk 1 above)
Comment (onacitarhan17) [reply src/main/java/model/crypto/Sha3256Hash.java]: fixed in https://github.com/yhassanzadeh13/lightchain-sdp/pull/40/commits/1393d8a8ae3fcb7fb95b93f740ef43adb1a5b549

File: src/main/java/model/crypto/Signature.java (hunk 2 above)
Comment (onacitarhan17) [reply src/main/java/model/crypto/Signature.java]: fixed in https://github.com/yhassanzadeh13/lightchain-sdp/pull/40/commits/1393d8a8ae3fcb7fb95b93f740ef43adb1a5b549

File: src/main/java/modules/ads/AuthenticatedEntity.java (hunk 3 above)
Comment (onacitarhan17) [reply src/main/java/modules/ads/AuthenticatedEntity.java]: fixed in https://github.com/yhassanzadeh13/lightchain-sdp/pull/40/commits/1393d8a8ae3fcb7fb95b93f740ef43adb1a5b549 (late reply)

**Review by thep2p — DISMISSED** [review]

**Review by thep2p — DISMISSED** [review]

**Pull request merged**